Author: Mike Markley <mike@markley.org>
Description: Fixes shebangs in examples (Closes: #152367).

diff -Naurp expect.orig/example/archie expect/example/archie
--- expect.orig/example/archie	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/archie	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 
 # archie
 
diff -Naurp expect.orig/example/autoexpect expect/example/autoexpect
--- expect.orig/example/autoexpect	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/autoexpect	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # Name: autoexpect - generate an Expect script from watching a session
 #
 # Description:
diff -Naurp expect.orig/example/autopasswd expect/example/autopasswd
--- expect.orig/example/autopasswd	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/autopasswd	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # wrapper to make passwd(1) be non-interactive
 # username is passed as 1st arg, passwd as 2nd
 
diff -Naurp expect.orig/example/beer.exp expect/example/beer.exp
--- expect.orig/example/beer.exp	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/beer.exp	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect -f
+#!/usr/bin/expect -f
 
 # 99 bottles of beer on the wall, Expect-style
 # Author: Don Libes <libes@nist.gov>
diff -Naurp expect.orig/example/chess.exp expect/example/chess.exp
--- expect.orig/example/chess.exp	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/chess.exp	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # expect script to connect two UNIX chess programs together.
 # written by Don Libes - May 9, 1990
 
diff -Naurp expect.orig/example/cryptdir expect/example/cryptdir
--- expect.orig/example/cryptdir	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/cryptdir	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # Name: cryptdir
 # Author: Don Libes, NIST
 #
diff -Naurp expect.orig/example/decryptdir expect/example/decryptdir
--- expect.orig/example/decryptdir	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/decryptdir	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # Name: cryptdir
 # Author: Don Libes, NIST
 #
diff -Naurp expect.orig/example/dislocate expect/example/dislocate
--- expect.orig/example/dislocate	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/dislocate	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # dislocate - allow disconnection and reconnection to a background program
 # Author: Don Libes, NIST
 
diff -Naurp expect.orig/example/dvorak expect/example/dvorak
--- expect.orig/example/dvorak	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/dvorak	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # simulate a dvorak keyboard
 # Actually just the lowercase letters are mapped to show the basic idea.
 # Really, uppercase and control should probably be mapped too.
diff -Naurp expect.orig/example/expectd.proto expect/example/expectd.proto
--- expect.orig/example/expectd.proto	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/expectd.proto	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/tcl/src/expect/e --
+#!/usr/bin/expectk -f
 # Description: Simple fragment to begin a telnet daemon
 # For more information, see Chapter 17 of "Exploring Expect"
 # Author: Don Libes, NIST
diff -Naurp expect.orig/example/ftp-inband expect/example/ftp-inband
--- expect.orig/example/ftp-inband	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/ftp-inband	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # ftp-inband - copy files over a telnet/rlogin/etc link
 # Author: Don Libes, NIST
 # Date: Jan 11, 1993
diff -Naurp expect.orig/example/ftp-rfc expect/example/ftp-rfc
--- expect.orig/example/ftp-rfc	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/ftp-rfc	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 
 # ftp-rfc <rfc-number>
 # ftp-rfc -index
diff -Naurp expect.orig/example/gethostbyaddr expect/example/gethostbyaddr
--- expect.orig/example/gethostbyaddr	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/gethostbyaddr	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 #
 # gethostbyaddr a.b.c.d - translate an internet address to a FQDN,
 #			  guessing (a lot) if necessary.
diff -Naurp expect.orig/example/irsh expect/example/irsh
--- expect.orig/example/irsh	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/irsh	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect --
+#!/usr/bin/expect --
 
 # Do rsh interactively.  For example, consider the following command:
 #    rsh <remote> ls -l "|" more
diff -Naurp expect.orig/example/kibitz expect/example/kibitz
--- expect.orig/example/kibitz	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/kibitz	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # allow another user to share a shell (or other program) with you
 # See kibitz(1) man page for complete info.
 # Author: Don Libes, NIST
diff -Naurp expect.orig/example/lpunlock expect/example/lpunlock
--- expect.orig/example/lpunlock	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/lpunlock	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 
 # This script unhangs a printer which claims it is "waiting for lock".
 # Written by Don Libes.  Based on English instructions from Scott Paisley.
diff -Naurp expect.orig/example/mkpasswd expect/example/mkpasswd
--- expect.orig/example/mkpasswd	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/mkpasswd	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect --
+#!/usr/bin/expect --
 # mkpasswd - make a password, if username given, set it.
 # Author: Don Libes, NIST
 
diff -Naurp expect.orig/example/multixterm expect/example/multixterm
--- expect.orig/example/multixterm	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/multixterm	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk
+#!/usr/bin/expectk
 #
 # NAME
 #	multixterm - drive multiple xterms separately or together
diff -Naurp expect.orig/example/passmass expect/example/passmass
--- expect.orig/example/passmass	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/passmass	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # passmass: change password on many machines
 # Synopsis: passmass host1 host2 host3 ....
 # Don Libes - March 11, 1991
diff -Naurp expect.orig/example/passwd.cgi expect/example/passwd.cgi
--- expect.orig/example/passwd.cgi	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/passwd.cgi	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect --
+#!/usr/bin/expect --
 
 # This is a CGI script to process requests created by the accompanying
 # passwd.html form.  This script is pretty basic, although it is
diff -Naurp expect.orig/example/passwdprompt expect/example/passwdprompt
--- expect.orig/example/passwdprompt	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/passwdprompt	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect
+#!/usr/bin/expect
 
 # This script prompts for a passwd from stdin while echoing *'s
 
diff -Naurp expect.orig/example/read1char expect/example/read1char
--- expect.orig/example/read1char	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/read1char	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 
 # read a single character
 # Author: Don Libes, NIST
diff -Naurp expect.orig/example/reprompt expect/example/reprompt
--- expect.orig/example/reprompt	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/reprompt	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect --
+#!/usr/bin/expect --
 
 # Name: reprompt
 # Description: reprompt every so often until user enters something
diff -Naurp expect.orig/example/rftp expect/example/rftp
--- expect.orig/example/rftp	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/rftp	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # rftp - ftp a directory hierarchy (i.e. recursive ftp)
 # Version 2.10
 # Don Libes, NIST
diff -Naurp expect.orig/example/rlogin-cwd expect/example/rlogin-cwd
--- expect.orig/example/rlogin-cwd	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/rlogin-cwd	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # rlogin-cwd - rlogin but with same directory
 #
 # You can extend this idea to save any arbitrary information across rlogin
diff -Naurp expect.orig/example/robohunt expect/example/robohunt
--- expect.orig/example/robohunt	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/robohunt	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # Synopsis
 #    robohunt player-name [-nodisplay]
 
diff -Naurp expect.orig/example/rogue.exp expect/example/rogue.exp
--- expect.orig/example/rogue.exp	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/rogue.exp	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # Look for a GREAT game of rogue.
 # Idea is that any game with a Strength of 18 is unusually good.
 # Written by Don Libes - March, 1990
diff -Naurp expect.orig/example/telnet-cwd expect/example/telnet-cwd
--- expect.orig/example/telnet-cwd	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/telnet-cwd	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # telnet-cwd - telnet but with same directory
 #
 # You can extend this idea to save any arbitrary information across telnet
diff -Naurp expect.orig/example/term_expect expect/example/term_expect
--- expect.orig/example/term_expect	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/term_expect	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk
+#!/usr/bin/expectk
 
 # Name: tkterm - terminal emulator using Expect and Tk text widget, v3.0
 # Author: Don Libes, July '94
diff -Naurp expect.orig/example/timed-read expect/example/timed-read
--- expect.orig/example/timed-read	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/timed-read	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # read a complete line from stdin
 # aborting after the number of seconds (given as an argument)
 # - Don Libes
diff -Naurp expect.orig/example/timed-run expect/example/timed-run
--- expect.orig/example/timed-run	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/timed-run	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 # run a program for a given amount of time
 # i.e. time 20 long_running_program
 
diff -Naurp expect.orig/example/tknewsbiff expect/example/tknewsbiff
--- expect.orig/example/tknewsbiff	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/tknewsbiff	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk
+#!/usr/bin/expectk
 
 # Name: tknewsbiff
 # Author: Don Libes
diff -Naurp expect.orig/example/tkpasswd expect/example/tkpasswd
--- expect.orig/example/tkpasswd	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/tkpasswd	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk -f
+#!/usr/bin/expectk -f
 # tkpasswd - Change passwords using Expectk
 # Author: Don Libes, NIST, October 1, 1993
 # Version: 1.8 - Added support for Tk 4.1
diff -Naurp expect.orig/example/tkterm expect/example/tkterm
--- expect.orig/example/tkterm	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/tkterm	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk
+#!/usr/bin/expectk
 
 # Name: tkterm - terminal emulator using Expect and Tk text widget, v3.0
 # Author: Don Libes, July '94
diff -Naurp expect.orig/example/unbuffer expect/example/unbuffer
--- expect.orig/example/unbuffer	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/unbuffer	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 # Description: unbuffer stdout of a program
 # Author: Don Libes, NIST
 
diff -Naurp expect.orig/example/virterm expect/example/virterm
--- expect.orig/example/virterm	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/virterm	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 
 # Name: virterm - terminal emulator using Expect, v1.0, December, 1994
 # Author: Adrian Mariano <adrian@cam.cornell.edu>
diff -Naurp expect.orig/example/vrfy expect/example/vrfy
--- expect.orig/example/vrfy	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/vrfy	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect -f
+#!/usr/bin/expect -f
 
 
 # separate address into user and host
diff -Naurp expect.orig/example/weather expect/example/weather
--- expect.orig/example/weather	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/weather	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect -f
+#!/usr/bin/expect -f
 
 # weather - Expect script to get the weather (courtesy University of Michigan)
 # Don Libes
diff -Naurp expect.orig/example/xkibitz expect/example/xkibitz
--- expect.orig/example/xkibitz	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/xkibitz	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!../expect --
+#!/usr/bin/expect --
 
 # share an xterm with other users
 # See xkibitz(1) man page for complete info.
diff -Naurp expect.orig/example/xpstat expect/example/xpstat
--- expect.orig/example/xpstat	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/xpstat	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expectk
+#!/usr/bin/expectk
 
 # This script acts as a front-end for xpilot.  Run it in the background,
 # and it will pop up a window for each server it finds running.  After
diff -Naurp expect.orig/example/xrlogin expect/example/xrlogin
--- expect.orig/example/xrlogin	2009-08-03 06:58:30.000000000 +0000
+++ expect/example/xrlogin	2009-08-03 08:31:41.000000000 +0000
@@ -1,4 +1,4 @@
-#!/depot/path/expect --
+#!/usr/bin/expect --
 # xrlogin - rlogin but with current DISPLAY
 #
 # You can extend this idea to save any arbitrary information across rlogin
