Author: John R.Daily <jdaily@progeny.com>
Description: Fixes problems on 64bit platforms (Closes: #126091).

diff -Naurp expect.orig/exp_main_tk.c expect/exp_main_tk.c
--- expect.orig/exp_main_tk.c	2009-08-03 06:58:30.000000000 +0000
+++ expect/exp_main_tk.c	2009-08-03 08:34:14.000000000 +0000
@@ -33,6 +33,7 @@ static char sccsid[] = "@(#) tkAppInit.c
 #endif /* not lint */
 
 #include <ctype.h>
+#include <string.h>  /* strrchr(3) */
 
 #include "tk.h"
 
diff -Naurp expect.orig/pty_termios.c expect/pty_termios.c
--- expect.orig/pty_termios.c	2009-08-03 06:58:30.000000000 +0000
+++ expect/pty_termios.c	2009-08-03 08:34:14.000000000 +0000
@@ -14,7 +14,9 @@ would appreciate credit if you use this 
 #define SIGCHLD SIGCLD
 #endif
 
+#include "expect.h"
 #include "expect_cf.h"
+#include "exp_int.h"  /* expErrnoMsg() prototype */
 
 /*
    The following functions are linked from the Tcl library.  They
