Author: n/a
Description: Fixes some unsafe traps.

diff -Naurp expect.orig/exp_trap.c expect/exp_trap.c
--- expect.orig/exp_trap.c	2009-08-03 06:58:30.000000000 +0000
+++ expect/exp_trap.c	2009-08-03 08:37:20.000000000 +0000
@@ -261,6 +261,18 @@ exp_init_trap()
 #if defined(SIGSTOP)
 	traps[SIGSTOP].reserved = TRUE;
 #endif
+#if defined(SIGSEGV)
+	traps[SIGSEGV].reserved = TRUE;
+#endif
+#if defined(SIGBUS)
+	traps[SIGBUS].reserved = TRUE;
+#endif
+#if defined(SIGILL)
+	traps[SIGILL].reserved = TRUE;
+#endif
+#if defined(SIGFPE)
+	traps[SIGFPE].reserved = TRUE;
+#endif
 
 	async_handler = Tcl_AsyncCreate(tophalf,(ClientData)0);
 
