Author: Adam Tkac <atkac@redhat.com>
Description: Fixes for tcl8.5.

diff -Naurp expect.orig/Dbg.c expect/Dbg.c
--- expect.orig/Dbg.c	2009-08-03 06:58:30.000000000 +0000
+++ expect/Dbg.c	2009-08-03 08:41:01.000000000 +0000
@@ -10,6 +10,7 @@ would appreciate credit if this program 
 
 #include <stdio.h>
 
+#include "expect_cf.h"
 #include "tcldbgcf.h"
 #if 0
 /* tclInt.h drags in stdlib.  By claiming no-stdlib, force it to drag in */
diff -Naurp expect.orig/exp_inter.c expect/exp_inter.c
--- expect.orig/exp_inter.c	2009-08-03 06:58:30.000000000 +0000
+++ expect/exp_inter.c	2009-08-03 08:41:01.000000000 +0000
@@ -36,6 +36,7 @@ would appreciate credit if this program 
 #include <ctype.h>
 
 #include "tcl.h"
+#include "tclInt.h"
 #include "string.h"
 
 #include "exp_tty_in.h"
