Author: Vitezslav Crhonek <vcrhonek@redhat.com>
Description: Fixes for new Tcl.

diff -Naurp expect.orig/exp_command.c expect/exp_command.c
--- expect.orig/exp_command.c	2009-08-03 09:23:29.000000000 +0000
+++ expect/exp_command.c	2009-08-03 09:24:31.000000000 +0000
@@ -2321,12 +2321,11 @@ Tcl_Obj *CONST objv[];	/* Argument objec
 	/* Historical note: we used "close"  long before there was a */
 	/* Tcl builtin by the same name. */
 
-	Tcl_CmdInfo info;
+	Tcl_CmdInfo* close_info;
 	Tcl_ResetResult(interp);
-	if (0 == Tcl_GetCommandInfo(interp,"close",&info)) {
-	    info.clientData = 0;
-	}
-	return(Tcl_CloseObjCmd(info.clientData,interp,objc_orig,objv_orig));
+
+	close_info = (Tcl_CmdInfo*) Tcl_GetAssocData (interp, EXP_CMDINFO_CLOSE, NULL);
+	return(close_info->objProc(close_info->objClientData,interp,objc,objv));
     }
 
     if (chanName) {
@@ -2971,7 +2970,10 @@ Tcl_Obj *CONST objv[];
     /* if successful (i.e., TCL_RETURN is returned) */
     /* modify the result, so that we will handle it specially */
 
-    int result = Tcl_ReturnObjCmd(clientData,interp,objc,objv);
+    Tcl_CmdInfo* return_info = (Tcl_CmdInfo*)
+    Tcl_GetAssocData (interp, EXP_CMDINFO_RETURN, NULL);
+
+    int result = return_info->objProc(return_info->objClientData,interp,objc,objv);
     if (result == TCL_RETURN)
         result = EXP_TCL_RETURN;
     return result;
diff -Naurp expect.orig/exp_command.h expect/exp_command.h
--- expect.orig/exp_command.h	2009-08-03 09:23:29.000000000 +0000
+++ expect/exp_command.h	2009-08-03 09:25:30.000000000 +0000
@@ -334,3 +334,6 @@ EXTERN int		Exp_StringCaseMatch _ANSI_AR
 EXTERN int		expSizeGet _ANSI_ARGS_((ExpState *));
 EXTERN int		expSizeZero _ANSI_ARGS_((ExpState *));
 #endif /* __EXP_COMMAND_H */
+
+#define EXP_CMDINFO_CLOSE  "expect/cmdinfo/close"
+#define EXP_CMDINFO_RETURN "expect/cmdinfo/return"
